src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -1085,17 +1085,6 @@ internal DirectiveStack Directives
         protected TNode CheckFeatureAvailability<TNode>(TNode node, MessageID feature, bool forceWarning = false)
             where TNode : GreenNode
         {
-            LanguageVersion availableVersion = this.Options.LanguageVersion;
-
-            // There are special error codes for some features, so handle those separately.
-            switch (feature)
-            {
-                case MessageID.IDS_FeatureModuleAttrLoc:
-                    return availableVersion >= LanguageVersion.CSharp2
-                        ? node
-                        : this.AddError(node, ErrorCode.WRN_NonECMAFeature, feature.Localize());
-            }
```

Comment:
one of the weirdest checks in the compiler.  It has been preserved, just with this logic moving to the exact place it needs to be checked.